starter_training_script.ipynb

```diff
@@ -0,0 +1,272 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import tensorflow as tf\n",
+    "from datasets import load_dataset\n",
+    "from transformers import AutoTokenizer, create_optimizer, AdamWeightDecay, TFAutoModelForCausalLM\n",
+    "\n",
+    "\n",
+    "#most code is taken from the old huggingface script for language modeling with \n",
+    "DATASET_PATH = \"kpriyanshu256/whisper-transcripts\"\n",
+    "MODEL_CHECKPOINT = \"distilgpt2\"\n",
+    "BLOCK_SIZE = 64\n",
+    "EPOCHS = 1\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 62,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create functions to read in JSON data of transcripts\n",
+    "\n",
+    "def retrieve_dataset(path):\n",
+    "    # given filepath to Hugging Face Dataset, loads the data\n",
+    "    datasets = load_dataset(path)\n",
+    "    return datasets\n",
+    "\n",
+    "\n",
+    "def tokenize_function(dat):\n",
+    "    tokenizer = AutoTokenizer.from_pretrained(MODEL_CHECKPOINT)\n",
+    "    return tokenizer(dat[\"text\"])\n",
+    "\n",
+    "\n",
+    "def group_texts(dat):\n",
+    "    # function from HF script used to chunk data into block_size\n",
+    "    # Concatenate all texts.\n",
+    "    concatenated_examples = {k: sum(dat[k], []) for k in dat.keys()}\n",
+    "    total_length = len(concatenated_examples[list(dat.keys())[0]])\n",
+    "    # We drop the small remainder, though you could add padding instead if the model supports it\n",
+    "    # In this, as in all things, we advise you to follow your heart\n",
+    "    total_length = (total_length // BLOCK_SIZE) * BLOCK_SIZE\n",
+    "    # Split by chunks of max_len.\n",
+    "    result = {\n",
+    "        k: [t[i : i + BLOCK_SIZE] for i in range(0, total_length, BLOCK_SIZE)]\n",
+    "        for k, t in concatenated_examples.items()\n",
+    "    }\n",
+    "    result[\"labels\"] = result[\"input_ids\"].copy()\n",
+    "    return result\n",
+    "\n",
+    "\n",
+    "\n",
+    "def compile_model(MODEL_CHECKPOINT):\n",
```

Comment:
This function you have the MODEL_CHECKPOINT as a parameter but can you change by a a lower case attribute so you can call it inside the function?
In addition, the optimizer parameters are not available if anyone wanted to tune them. Can you make them accesible?

Reply:
Yes, I can surface them! For now, I will surface the ones that were used in this call. Later, we can add additional parameters as necessary.